src/emulator/storage/apis/firebase.ts

```diff
@@ -323,186 +327,159 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
         }
         throw err;
       }
-      const metadata = JSON.parse(metadataRaw)!;
-      const md = storageLayer.oneShotUpload(
-        req.params.bucketId,
-        name,
-        metadata.contentType!,
-        metadata,
-        dataRaw
-      );
-
-      if (!md) {
-        res.sendStatus(400);
-        return;
-      }
-
-      const operationPath = ["b", req.params.bucketId, "o", name].join("/");
-
-      if (
-        !(await isPermitted({
-          ruleset: emulator.rules,
-          // TODO: This will be either create or update
-          method: RulesetOperationMethod.CREATE,
-          path: operationPath,
-          authorization: req.header("authorization"),
-          file: {
-            after: md?.asRulesResource(),
-          },
-        }))
-      ) {
-        storageLayer.deleteFile(md?.bucket, md?.name);
-        return res.status(403).json({
-          error: {
-            code: 403,
-            message: `Permission denied. No WRITE permission.`,
-          },
-        });
-      }
-
-      if (md.downloadTokens.length === 0) {
-        md.addDownloadToken();
+      const upload = uploadService.multipartUpload({
+        bucketId,
+        objectId,
+        metadataRaw,
```

Comment:
@tohhsinpei actually mentioned it may be more idiomatic to do it this way. I looked through the style guide and it wasn't mentioned. @sam-gc do you know which style is preferred?

Reply:
They're not mentioned in the TS styleguide but they are in the JS styleguide: https://google.github.io/styleguide/jsguide.html#features-objects-shorthand-properties (only insofar as they're allowed. No preference given).

Personally, in all the typescript projects I've worked on, we actively encourage object shorthand where applicable, even if some properties _are not_ in shorthand. Is object shorthand used elsewhere in this repo?

Reply:
Gotcha yeah don't have a strong opinion since I didn't find it in the style guide as well. Only reasons I can think of are

1. Maintain consistency with the rest of the file
2. On the off chance variable names change or we're passing in some other value in the future, it will still be explicit what that field is.

Reply:
FWIW I don't see too many other instances where we could use shorthand in this file. Re: your second point, object shorthand is only applicable when the variable name matches the property name; it's not based on arg position.

for some type:
```
type Point { x: number, y: number}
```

This is valid:
```
let p: Point = { y, x }
```

This is invalid
```
let p: Point = { someX, someRandomThing }
```

Reply:
2 would be caught by the compiler/type checker. I'm pinging you both a note about this

Reply:
Ahh I see, sorry for the confusion. In that case shorthand seems to be the way to go.

Reply:
No problem, I agree let's bias to using shorthand moving forward. Glad we got clarity on this, thanks all